CMP-3969, CMP-4121: Fix panic issue when deleting a scan with invalid storage size

xiaojiey opened the pull request:
Description of the PR:
1. Fix panic issue when storage size is invalid
2. Add CEL validation to RawResultStorageSettings.Size field to validate quantity format at API creation time using isQuantity() function
3. Add e2e test to verify invalid storage sizes are rejected at API level

xiaojiey commented:
/retest-required

rhmdnd commented on pkg/controller/compliancescan/rawresults.go:
```diff
@@ -51,6 +51,21 @@ func (r *ReconcileComplianceScan) handleRawResultsForScan(instance *compv1alpha1
+	// Validate storage size before attempting to construct PVC
+	// If the size is invalid, we can't construct a valid PVC object, so log a warning and skip deletion.
+	// The PVC may not exist anyway if the scan failed validation before PVC creation.
+	if instance.Spec.RawResultStorage.Size != "" {
```
Nice catch - is this already covered in the e2e tests somewhere?

xiaojiey commented:
/retest

xiaojiey commented on tests/e2e/parallel/main_test.go:
```diff
@@ -5249,3 +5249,79 @@ func TestRuleVariableAnnotation(t *testing.T) {
+		t.Fatalf("failed to get ComplianceSuite: %s", err)
+	}
+
+	if suite.Status.Result != compv1alpha1.ResultError {
```
No, it is not related. The metrics check comes from downstream test case 43072, tracked with Jira ticket https://issues.redhat.com/browse/CMP-3829

xiaojiey commented:
/retest

rhmdnd reviewed (commented):
@yuumasato @Vincent056 what are your thoughts on this particular fix?

xiaojiey commented:
/retest

xiaojiey committed:
CMP-4147: Add CEL validation for RawResultStorage size

Add kubebuilder validation markers with CEL rules to validate that
RawResultStorage.Size is a valid Kubernetes quantity format. This
prevents invalid storage sizes (e.g., "1B", "1GB", "invalid-size")
from being accepted by the API, providing early feedback to users.

The validation applies to ComplianceScan, ComplianceSuite, and
ScanSetting CRDs. The CEL rule allows empty string or validates
the format using the built-in isQuantity() function.

Also add runtime safety check for deleting scans with invalid storage
sizes. This handles upgrade scenarios where existing scans may have
invalid sizes from before the validation was added. The deletion
operation will skip PVC deletion but allow the scan deletion to proceed.

Add e2e test to verify API-level rejection of invalid storage sizes.
